### DISCUSSION TITLE
cublas library path issues

### DISCUSSION THREAD

**Discussion by lfdmn** [OE4T/meta-tegra#533] [Q&A] [answered]
Hi,

Bumping yocto branch from zeus to dunfell. 

```libcublas``` is now packaged differently.

All the CUDA libraries get deployed in ```.../recipe-sysroot/usr/local/cuda-10.2/lib``` while cublas now ends up in ```.../recipe-sysroot/usr/lib/```.

I'm building a project (arrayfire) which is using ```cuda_find_library_local_first```  to find the libraries. It now fails on ```cublas``` and ```cublasLt```

Here's the cmake snippet ```cuda_find_library_local_first```
```cmake
macro(cuda_find_library_local_first_with_path_ext _var _names _doc _path_ext )
  if(CMAKE_SIZEOF_VOID_P EQUAL 8)
    # CUDA 3.2+ on Windows moved the library directories, so we need the new
    # and old paths.
    set(_cuda_64bit_lib_dir "${_path_ext}lib/x64" "${_path_ext}lib64" "${_path_ext}libx64" )
  endif()
  # CUDA 3.2+ on Windows moved the library directories, so we need to new
  # (lib/Win32) and the old path (lib).
  find_library(${_var}
    NAMES ${_names}
    PATHS "${CUDA_TOOLKIT_TARGET_DIR}"
    ENV CUDA_PATH
    ENV CUDA_LIB_PATH
    PATH_SUFFIXES ${_cuda_64bit_lib_dir} "${_path_ext}lib/Win32" "${_path_ext}lib" "${_path_ext}libWin32"
    DOC ${_doc}
    NO_DEFAULT_PATH
    )
  if (NOT CMAKE_CROSSCOMPILING)
    # Search default search paths, after we search our own set of paths.
    find_library(${_var}
      NAMES ${_names}
      PATHS "/usr/lib/nvidia-current"
      DOC ${_doc}
      )
  endif()
endmacro()

macro(cuda_find_library_local_first _var _names _doc)
  cuda_find_library_local_first_with_path_ext( "${_var}" "${_names}" "${_doc}" "" )
endmacro()
```

Since ```CUDA_TOOLKIT_TARGET_DIR=.../recipe-sysroot/usr/local/cuda-10.2``` libraries are not found.

Was this change intentional or fixing some other issues?

**Comment by madisongh** [comment, marked as answer]
> Was this change intentional or fixing some other issues?

You'd have to ask NVIDIA that question :) - they're the ones that made that change (in CUDA 10.1, IIRC).  We're just following what they've done.  Any project with home-grown CUDA discovery in their build system has to be updated to account for the change.

**Reply by lfdmn** [reply]
Understood =)
